.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,42 @@
+name: "Build and push image"
+on:
+  push:
+    branches: master
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: QEmu setup
+        uses: actions/checkout@v2
+
+      - name: Docker buildx setup
+        uses: docker/setup-buildx-action@v1
+
+      - name: GHCR access credentials
+        id: ghcr
+        uses: hashicorp/vault-action@v2.0.1
+        with:
+          url: https://volt.cap.explore.suse.dev/ui/vault/secrets
+          method: github
+          githubToken: ${{ secrets.VaultToken }}
+          caCertificate: ${{ secrets.VaultCert }}
+          secrets: |
+            secret/show/kubecf ghcr-user     | GHCR_USER
+            secret/show/kubecf ghcr-password | GHCR_PASSWORD
+
+      - name: GHCR login
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ steps.ghcr.outputs.GHCR_USER }}
+          password: ${{ steps.ghcr.outputs.GHCR_PASSWORD }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
```

Comment:
Yes, we probably want _both_ tags and labels.  A docker image, `ghcr.io/org/name:tag` has a tag; each image (as in the bits, in a tar archive) can have labels that you can introspect after pulling (so it's easier to tell that the image came from this git repo).

Now that we have the labels, we want to figure out what the tag should be too (so that we can pull the image afterwards, consistently), right?  I assume it pushes to `latest` by default, which is fine for development but not great for using it in other things (like KubeCF).

It looks like the concourse pipeline used the [semver resource](https://github.com/concourse/semver-resource), which is hard to duplicate here; we'll probably want to use the code from [kubecf-tools](https://github.com/cloudfoundry-incubator/kubecf-tools/tree/main/versioning) here.

(This only makes sense because this is the pipeline for `master`, so we want to actually publish the images for consumption.)